VideoLocker/src/main/java/org/edx/mobile/view/LaunchActivity.java

```diff
@@ -1,93 +1,76 @@
+                finish();
+            }
 
-        //Register for Login Receiver before checking for logged in user
-        enableLoginCallback();
-
-        //We need to stop the Launch Activity from launching if the user has logged in
-        PrefManager pm =new PrefManager(LaunchActivity.this, PrefManager.Pref.LOGIN);
-        if (pm.getCurrentUserProfile() != null) {
-            finish();
-            return;
-        }
-        setContentView(R.layout.activity_launch);
-
-        Button sign_in_tv = (Button) findViewById(R.id.sign_in_tv);
-        sign_in_tv.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onError(@NonNull Throwable error) {
+                // This will never happen
+                throw new RuntimeException(error);
+            }
+        });
+        final ActivityLaunchBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_launch);
+        binding.signInTv.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
                 environment.getRouter().showLogin(LaunchActivity.this);
             }
         });
-
-        Button sign_up_button = (Button) findViewById(R.id.sign_up_btn);
-        sign_up_button.setOnClickListener(new OnClickListener() {
+        binding.signUpBtn.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
-                try {
-                    environment.getSegment().trackUserSignUpForAccount();
-                }catch(Exception e){
-                    logger.error(e);
-                }
+                environment.getSegment().trackUserSignUpForAccount();
                 environment.getRouter().showRegistration(LaunchActivity.this);
             }
         });
-
         environment.getSegment().trackScreenView(ISegment.Screens.LAUNCH_ACTIVITY);
     }
 
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        disableLoginCallback();
-    }
-
-    //Broadcast Receiver to notify all activities to finish if user logs out
-    private BroadcastReceiver loginReceiver = new BroadcastReceiver() {
-        @Override
-        public void onReceive(Context context, Intent intent) {
-            finish();
-        }
-    };
-
-    protected void enableLoginCallback() {
-        // register for login listener
-        IntentFilter filter = new IntentFilter();
-        filter.addAction(AppConstants.USER_LOG_IN);
-        registerReceiver(loginReceiver, filter);
-    }
-
-    protected void disableLoginCallback() {
-        // un-register loginReceiver
-        unregisterReceiver(loginReceiver);
+        logInSubscription.unsubscribe();
     }
 
-
-
     @Override
     protected boolean createOptionsMenu(Menu menu) {
-        // Launch screen doesn't have any menu
-        return true;
+        return false; // Disable menu inherited from BaseFragmentActivity
```

Comment:
Because our base activity class `BaseFragmentActivity` has an option menu by default and we want to suppress it.

On my long list of things to clean up if I have time: https://openedx.atlassian.net/browse/MA-2093?filter=-1